src/Control/FunFlow/Steps.hs

```diff
@@ -64,3 +69,33 @@ retry n secs f = catch f $ proc (x, (_ :: ex)) -> do
   x1 <- pauseWith -< (secs,x)
   x2 <- retry (n-1) secs f -< x1
   returnA -< x2
+
+-- | @putFileInStore -< (inFile, outFile)@ copies the contents of @inFile@
+-- into the content store under the file name @outFile@.
+putFileInStore :: Flow eff ex (FileContent, FilePath) CS.Item
+putFileInStore = putInStore $ \dir (FileContent inPath, outFile) -> do
+  let outPath = dir </> outFile
+  copyFile inPath outPath
+
+-- | @putDirInStore -< (inDir, Just outDir)@ copies the contents of @inDir@
+-- into the content store under @outDir@.
+-- @putDirInStore -< (inDir, Nothing)@ copies the contents of @inDir@
+-- into the content store directly under the subtree.
+putDirInStore :: Flow eff ex (DirectoryContent, Maybe FilePath) CS.Item
```

Comment:
I wonder whether we should look at using the `path` library. Might have a few advantages:

- Validation of files/directories
- Can restrict to e.g. relative dirs when accessing things from the store.